e2etest/newe2e_resource_manager_azstorage.go

```diff
@@ -30,47 +32,88 @@ func (acct *AzureAccountResourceManager) ApplySAS(URI string, loc common.Locatio
 		return URI
 	}
 
-	sasVals := opts.AzureOpts.SASValues
-	if sasVals == nil {
+	var sasVals GenericSignatureValues
+	if opts.AzureOpts.SASValues == nil {
+		// Default to account level SAS to cover all our bases
+		sasVals = GenericAccountSignatureValues{}
+	} else {
+		sasVals = opts.AzureOpts.SASValues
 
-		sasVals = GenericServiceSignatureValues{}
 	}
 
 	switch loc {
 	case common.ELocation.Blob():
-		skc, err := blobservice.NewSharedKeyCredential(acct.accountName, acct.accountKey)
+		parts, err := blobsas.ParseURL(URI)
 		common.PanicIfErr(err)
 
-		p, err := sasVals.AsBlob().SignWithSharedKey(skc)
+		skc, err := blobservice.NewSharedKeyCredential(acct.accountName, acct.accountKey)
 		common.PanicIfErr(err)
 
-		parts, err := blobsas.ParseURL(URI)
+		var p blobsas.QueryParameters
+		if _, ok := sasVals.(GenericServiceSignatureValues); ok {
+			v := sasVals.(GenericServiceSignatureValues)
+			v.Level = common.Iff(v.Level == nil, to.Ptr(lev), v.Level)
+			v.ContainerName = parts.ContainerName
+			v.ObjectName = parts.BlobName
+
+			p, err = v.AsBlob().SignWithSharedKey(skc)
+		} else {
+			p, err = sasVals.AsBlob().SignWithSharedKey(skc)
+		}
 		common.PanicIfErr(err)
 
 		parts.SAS = p
 		parts.Scheme = common.Iff(opts.RemoteOpts.Scheme != "", opts.RemoteOpts.Scheme, "https")
 		return parts.String()
 	case common.ELocation.File():
-		skc, err := fileservice.NewSharedKeyCredential(acct.accountName, acct.accountKey)
+		parts, err := filesas.ParseURL(URI)
 		common.PanicIfErr(err)
 
-		p, err := sasVals.AsFile().SignWithSharedKey(skc)
+		skc, err := fileservice.NewSharedKeyCredential(acct.accountName, acct.accountKey)
 		common.PanicIfErr(err)
 
-		parts, err := filesas.ParseURL(URI)
+		var p filesas.QueryParameters
+		if _, ok := sasVals.(GenericServiceSignatureValues); ok {
+			v := sasVals.(GenericServiceSignatureValues)
+			v.Level = common.Iff(v.Level == nil, to.Ptr(lev), v.Level)
+			v.ContainerName = parts.ShareName
+			if et == common.EEntityType.Folder() {
+				v.DirectoryPath = parts.DirectoryOrFilePath
+			} else {
+				v.ObjectName = parts.DirectoryOrFilePath
+			}
+
+			p, err = v.AsFile().SignWithSharedKey(skc)
+		} else {
+			p, err = sasVals.AsFile().SignWithSharedKey(skc)
+		}
 		common.PanicIfErr(err)
 
 		parts.SAS = p
 		parts.Scheme = common.Iff(opts.RemoteOpts.Scheme != "", opts.RemoteOpts.Scheme, "https")
 		return parts.String()
 	case common.ELocation.BlobFS():
-		skc, err := blobfscommon.NewSharedKeyCredential(acct.accountName, acct.accountKey)
+		parts, err := datalakesas.ParseURL(URI)
 		common.PanicIfErr(err)
 
-		p, err := sasVals.AsDatalake().SignWithSharedKey(skc)
+		skc, err := blobfscommon.NewSharedKeyCredential(acct.accountName, acct.accountKey)
 		common.PanicIfErr(err)
 
-		parts, err := datalakeSAS.ParseURL(URI)
+		var p datalakesas.QueryParameters
+		if _, ok := sasVals.(GenericServiceSignatureValues); ok {
```

Comment:
(nit) A thought for condensing this: Have an anonymous function above that appends this information? Something like `setupCanon(containerName, objectPath string, isDir bool)`

Reply:
will try this

Reply:
irrelevant now